src/xcode/ENA/ENA/Source/Scenes/Events/OnBehalfCheckinSubmission/TraceLocationSelection/OnBehalfTraceLocationSelectionViewController.swift

```diff
@@ -0,0 +1,246 @@
+		self.onScanQRCodeCellTap = onScanQRCodeCellTap
+		self.onMissingPermissionsButtonTap = onMissingPermissionsButtonTap
+		self.onPrimaryButtonTap = onPrimaryButtonTap
+		self.onDismiss = onDismiss
+		
+		super.init(style: .plain)
+	}
+	
+	required init?(coder: NSCoder) {
+		fatalError("init(coder:) has not been implemented")
+	}
+	
+	// MARK: - Overrides
+	
+	override func viewDidLoad() {
+		super.viewDidLoad()
+
+		parent?.title = viewModel.title
+		parent?.navigationItem.largeTitleDisplayMode = .always
+		parent?.navigationItem.rightBarButtonItem = dismissHandlingCloseBarButton
+
+		setUpTableView()
+
+		tableView.reloadData()
+		setUpEmptyState()
+		
+		viewModel.$continueEnabled
+			.receive(on: DispatchQueue.main.ocombine)
+			.sink { [weak self] in
+				self?.footerView?.setEnabled($0, button: .primary)
+			}
+			.store(in: &subscriptions)
+
+		viewModel.$triggerReload
+			.receive(on: DispatchQueue.main.ocombine)
+			.sink { [weak self] _ in
+				guard let self = self else { return }
+
+				self.tableView.reloadData()
+				self.setUpEmptyState()
+			}
+			.store(in: &subscriptions)
+	}
+
+	override func viewWillAppear(_ animated: Bool) {
+		super.viewWillAppear(animated)
+
+		navigationController?.navigationBar.prefersLargeTitles = true
+		navigationController?.navigationBar.sizeToFit()
+	}
+	
+	// MARK: - Protocol DismissHandling
+	
+	func wasAttemptedToBeDismissed() {
+		onDismiss()
+	}
+		
+	// MARK: - Protocol FooterViewHandling
+	
+	func didTapFooterViewButton(_ type: FooterViewModel.ButtonType) {
+		guard type == .primary, let selectedTraceLocation = viewModel.selectedTraceLocation else {
+			return
+		}
+
+		onPrimaryButtonTap(selectedTraceLocation)
+	}
+	
+	// MARK: - Protocol UITableViewDataSource
+
+	override func numberOfSections(in tableView: UITableView) -> Int {
+		return viewModel.numberOfSections
+	}
+
+	override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+		return viewModel.numberOfRows(in: section)
+	}
+
+	override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+		switch OnBehalfTraceLocationSelectionViewModel.Section(rawValue: indexPath.section) {
+		case .description:
+			return descriptionCell(forRowAt: indexPath)
+		case .qrCodeScan:
+			return scanTraceLocationCell(forRowAt: indexPath)
+		case .missingCameraPermission:
+			return missingPermissionsCell(forRowAt: indexPath)
+		case .traceLocations:
+			return traceLocationCell(forRowAt: indexPath)
+		default:
+			fatalError("Invalid section")
+		}
+	}
+
+	// MARK: - Protocol UITableViewDelegate
+	
+	override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+		switch OnBehalfTraceLocationSelectionViewModel.Section(rawValue: indexPath.section) {
+		case .description:
+			return
+		case .qrCodeScan:
+			onScanQRCodeCellTap()
+		case .missingCameraPermission:
+			return
+		case .traceLocations:
+			viewModel.toggleSelection(at: indexPath.row)
```

Comment:
Also here, I wonder if the indirection over the viewmodel and combine is really needed. Or could it be done easier by just calling an toggle on the cell.

Reply:
I think it would be possible to keep the selection state exclusively on the `UITableViewCell`, but I think the upside of this approach is that we have it testable in the model layer and don't need to rely on the view layer to provide us the selected trace locations but rather have the model as the source of truth.

Reply:
The trace locations could still be part of the view model. And then call update on a cell and inject the cell model from the view model.